src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -2723,7 +2726,9 @@ public DocCommentTree getDocCommentTree0(Element element) {
                     }
                 }
                 // run doclint even if docCommentTree is null, to trigger checks for missing comments
-                configuration.runDocLint(path);
+                if (shouldRunDocLint(element)) {
+                    configuration.runDocLint(path);
+                }
```

Comment:
I'm disappointed there is so much disruption to the code because of the enhanced treatment of `@hidden`.Maybe it's necessary, but I was hoping/expecting that we could add a separate code path for the specific purpose of seeing if a non-selected(?) class has a `@hidden` tag. 

For example, leave all the code for block tags and doc comment tree alone unchanged, and then provide `isHidden(Element)`  by just using `getDocCommentInfo`, which gives cached access to the `DocCommentTree` then scan the block tags for a `HIDDEN` tree.